src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaOauthSecurityConfig.java

```diff
@@ -0,0 +1,236 @@
+      httpSecurity.antMatcher("/api/**").headers().frameOptions().disable();
+      httpSecurity.antMatcher("/api/**").csrf().requireCsrfProtectionMatcher(new AntPathRequestMatcher("/api/oauth/authorize"))
+        .disable();
+      httpSecurity.antMatcher("/api/**").csrf().disable();
+      httpSecurity.antMatcher("/api/**").exceptionHandling().accessDeniedPage("/login?error");
+
+      // SecurityContextPersistenceFilter appears pretty high up (well
+      // before any OAuth related filters), so we'll put our anonymous
+      // user filter into the filter chain after that.
+      httpSecurity.antMatcher("/api/**").addFilterAfter(new UnauthenticatedAnonymousAuthenticationFilter("anonymousTokenAuthProvider"),
+        SecurityContextPersistenceFilter.class);
+    }
+  }
+
+  /**
+   * Class for configuring the OAuth authorization server
+   */
+  @Configuration
+  @EnableAuthorizationServer
+  protected static class AuthorizationServerConfiguration extends AuthorizationServerConfigurerAdapter {
+    @Autowired
+    @Qualifier("clientDetails")
+    private ClientDetailsService clientDetailsService;
+
+    @Autowired
+    @Qualifier("iridaTokenStore")
+    private TokenStore tokenStore;
+
+    @Autowired
+    @Qualifier("userAuthenticationManager")
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private WebResponseExceptionTranslator exceptionTranslator;
+
+    @Override
+    public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
+      clients.withClientDetails(clientDetailsService);
+    }
+
+    @Autowired
+    private ResourceServerTokenServices tokenServices;
+
+    @Override
+    public void configure(final AuthorizationServerEndpointsConfigurer endpoints) throws Exception {
+      endpoints.tokenStore(tokenStore);
+      endpoints.authenticationManager(authenticationManager);
+      endpoints.authorizationCodeServices(authorizationCodeServices());
+      endpoints.pathMapping("/oauth/token", "/api/oauth/token").allowedTokenEndpointRequestMethods(HttpMethod.GET, HttpMethod.POST);
+      endpoints.pathMapping("/oauth/check_token", "/api/oauth/check_token");
+      endpoints.pathMapping("/oauth/confirm_access", "/api/oauth/confirm_access");
+      endpoints.pathMapping("/oauth/error", "/api/oauth/error");
+      endpoints.pathMapping("/oauth/authorize", "/api/oauth/authorize");
+      endpoints.tokenServices((DefaultTokenServices)tokenServices);
+      endpoints.exceptionTranslator(exceptionTranslator);
+      endpoints.redirectResolver(new OauthRedirectResolver());
+    }
+
+    @Override
+    public void configure(final AuthorizationServerSecurityConfigurer oauthServer) throws Exception {
+      oauthServer
+        .tokenKeyAccess("permitAll()")
+        .checkTokenAccess("isAuthenticated()")
+        .allowFormAuthenticationForClients();
+      /*oauthServer.passwordEncoder(new PasswordEncoder() {
```

Comment:
I had copied this from your attempts at spring upgrade, I'm not sure if the password encoder is needed as we don't have a separate login endpoint for the api now.